shmem.c

```diff
@@ -120,11 +123,58 @@ void addOverTimeClientSlot() {
 	}
 }
 
+/// Create a read/write lock
+pthread_rwlock_t create_rwlock() {
+	pthread_rwlockattr_t lock_attr = {};
+	pthread_rwlock_t lock = {};
+
+	// Initialize the lock attributes
+	pthread_rwlockattr_init(&lock_attr);
+
+	// Allow the lock to be used by other processes
+	pthread_rwlockattr_setpshared(&lock_attr, PTHREAD_PROCESS_SHARED);
+
+	// Initialize the lock
+	pthread_rwlock_init(&lock, &lock_attr);
+
+	return lock;
+}
+
+void shm_read_lock() {
```

Comment:
We should a comment like: If a writer holds the lock, the calling thread cannot acquire the read lock. It blocks (that is, it does not return from `shm_read_lock()`) until the read lock is acquired. The benefit is that **multiple readers can access the data at the same time**.